pkg/apis/nsx.vmware.com/v1alpha1/vpcnetworkconfiguration_types.go

```diff
@@ -19,38 +24,41 @@ const (
 type VPCNetworkConfigurationSpec struct {
-	// PolicyPath of Tier0 or Tier0 VRF gateway.
-	DefaultGatewayPath string `json:"defaultGatewayPath,omitempty"`
-	// Edge cluster path on which the networking elements will be created.
-	EdgeClusterPath string `json:"edgeClusterPath,omitempty"`
 	// NSX-T Project the Namespace associated with.
-	NSXTProject string `json:"nsxtProject,omitempty"`
-	// NSX-T IPv4 Block paths used to allocate external Subnets.
-	// +kubebuilder:validation:MinItems=0
-	// +kubebuilder:validation:MaxItems=5
-	ExternalIPv4Blocks []string `json:"externalIPv4Blocks,omitempty"`
-	// Private IPv4 CIDRs used to allocate Private Subnets.
-	// +kubebuilder:validation:MinItems=0
-	// +kubebuilder:validation:MaxItems=5
-	PrivateIPv4CIDRs []string `json:"privateIPv4CIDRs,omitempty"`
-	// Default size of Subnet based upon estimated workload count.
-	// Defaults to 26.
-	// +kubebuilder:default=26
-	DefaultIPv4SubnetSize int `json:"defaultIPv4SubnetSize,omitempty"`
-	// DefaultSubnetAccessMode defines the access mode of the default SubnetSet for PodVM and VM.
-	// Must be Public or Private.
-	// +kubebuilder:validation:Enum=Public;Private
-	DefaultSubnetAccessMode string `json:"defaultSubnetAccessMode,omitempty"`
+	NsxProject string `json:"nsxProject,omitempty"`
+
+	// VpcConnectivityProfile ID. This profile has configuration related to create VPC transit gateway attachment.
+	VpcConnectivityProfile string `json:"vpcConnectivityProfile,omitempty"`
+
+	// The path of the configuration profile of the VPC services. This will be an collection of default dhcp and subnet profiles. The default vpc service profile will be created as part of new project create workflow. That will be used as the default for all VPC created under that project. The default value will be project specific default VPC profile.
+	VpcServiceProfile string `json:"vpcServiceProfile,omitempty"`
+
+	// Private IPs.
+	PrivateIPs []string `json:"privateIPs,omitempty"`
+
 	// ShortID specifies Identifier to use when displaying VPC context in logs.
-	// Less than or equal to 8 characters.
+	// Less than equal to 8 characters.
 	// +kubebuilder:validation:MaxLength=8
 	// +optional
 	ShortID string `json:"shortID,omitempty"`
+
 	// NSX path of the VPC the Namespace associated with.
 	// If vpc is set, only defaultIPv4SubnetSize and defaultSubnetAccessMode
 	// take effect, other fields are ignored.
 	// +optional
 	VPC string `json:"vpc,omitempty"`
+
+	// +kubebuilder:validation:Enum=SMALL;MEDIUM;LARGE;XLARGE
+	LbServiceSize string `json:"lbServiceSize,omitempty"`
+
+	// Default size of Subnet based upon estimated workload count.
+	// Defaults to 26.
+	// +kubebuilder:default=26
+	DefaultSubnetSize int `json:"defaultSubnetSize,omitempty"`
+	// PodSubnetAccessMode defines the access mode of the default SubnetSet for PodVM.
+	// Must be Public or Private.
+	// +kubebuilder:validation:Enum=Public;Private;Project
+	PodSubnetAccessMode string `json:"podSubnetAccessMode,omitempty"`
```

Comment:
It is still in discussion whether we should remove this one.

Reply:
OK, noted

Reply:
@jianjuns If it's removed, default will be `PrivateTGW` always right?

Reply:
Yes